src/select/types.js

```diff
@@ -0,0 +1,23 @@
+// @flow
+
+type ReactSelectValue = Object | Object[];
```

Comment:
We know the structure for this, right? Instead of using `Object`, why not something like this:

```
type ReactSelectOption = {|
  label :string;
  value :string;
|};
type ReactSelectValue = ReactSelectOption | ReactSelectOption[];
```